app/controllers/users_controller.rb

```diff
@@ -1,5 +1,10 @@
 class UsersController < ApplicationController
-  def index; end
+  def index
+    @users = User.all
+  end
 
-  def show; end
+  def show
+    @user = User.find(params[:id])
+    @posts = @user.recent_posts
+  end
```

Comment:
I couldn't test your application because it was not running. 

If you are using alias in your migration, you should specify it in your model also.
![Screenshot 2021-12-11 at 20-36-00 Action Controller Exception caught](https://user-images.githubusercontent.com/1642014/145686252-05529ab1-bc9c-4d1f-bd2a-b84731f3be7b.png)

For example
```ruby
has_many :posts, foreign_key: "author_id", class_name: "Post"
```

Reply:
Hi @sinansevgi👋, can you please shed more light on this🙏🏽 I don't get it.

Reply:
You are using author_id for user_id in your migration file but your model doesn't know this and checks for the user_id field in your posts table therefore it throws an error saying posts.user_id doesn't exist. You should configure your models to look for author_id instead of user_id